tests/template_tests/test_loaders.py

```diff
@@ -49,11 +49,46 @@ def test_get_template(self):
         self.assertEqual(template.origin.template_name, 'index.html')
         self.assertEqual(template.origin.loader, self.engine.template_loaders[0].loaders[0])
 
-    def test_get_template_missing(self):
+    def test_get_template_missing_debug_off(self):
+        """
+        With template debugging disabled, the raw TemplateDoesNotExist class
+        should be cached when a template is missing. See ticket #26306 and
+        docstrings in the cached loader for details.
+        """
+        self.engine.debug = False
         with self.assertRaises(TemplateDoesNotExist):
-            self.engine.get_template('doesnotexist.html')
-        e = self.engine.template_loaders[0].get_template_cache['doesnotexist.html']
-        self.assertEqual(e.args[0], 'doesnotexist.html')
+            self.engine.get_template('prod-template-missing.html')
+        e = self.engine.template_loaders[0].get_template_cache['prod-template-missing.html']
+        self.assertEqual(e, TemplateDoesNotExist)
+
+    def test_get_template_missing_debug_on(self):
+        """
+        With template debugging enabled, ensure that a TemplateDoesNotExist
+        instance is cached when a template is missing.
+        """
+        self.engine.debug = True
+        with self.assertRaises(TemplateDoesNotExist):
+            self.engine.get_template('debug-template-missing.html')
+        e = self.engine.template_loaders[0].get_template_cache['debug-template-missing.html']
+        self.assertIsInstance(e, TemplateDoesNotExist)
+        self.assertEqual(e.args[0], 'debug-template-missing.html')
+
+    @unittest.skipIf(six.PY2, "Python 2 doesn't set extra exception attributes")
+    def test_cached_exception_no_traceback(self):
+        """
+        Ensure the exception cached when template debugging is enabled doesn't
+        contain the traceback or context information that Python sets when an
+        exception is raised in Python 3.
```

Comment:
If you don't mention "Python 3" then that's one less change to make when dropping Python 2.